docs: add warning around using logstash plugin for new deployments (#14058)
grafana/loki: docs/sources/send-data/logstash/_index.md

`logstash-output-loki` that enables shipping logs to a Loki
instance or [Grafana Cloud](/products/cloud/).

{{< admonition type="warning" >}}
Grafana Labs does not recommend using the Logstash plugin for new deployments. Even as a mechanism for quickly testing Loki with your existing Beats/Logstash infrastructure we highly discourage the use of this plugin.

Our experience over the years has found numerous significant challenges using Logstash and this plugin:

  * It is very difficult to configure labels correctly.  Conceptually Elasticsearch is a very different database from Loki and users almost always end up sending too many high cardinality labels to Loki, which makes getting started with Loki unnecessarily complicated and confusing vs. using other clients.
  * Logstash and the upstream Beats components implement backoff and flow control which we've found hard to observe, leading to ingestion delays into Loki which are extremely difficult to address.

>   * We at Grafana Labs have no expertise at configuring Logstash or understanding its configuration language, so we cannot provide support for it.

Reviewer: ```suggestion
  * We at Grafana Labs have no expertise at configuring Logstash or understanding of its configuration language, so we cannot provide support for it.
```

`logstash-output-loki` that enables shipping logs to a Loki
instance or [Grafana Cloud](/products/cloud/).

> {{% admonition type="warning" %}}

Reviewer: ```suggestion
{{< admonition type="warning" >}}
```

> The Logstash plugin is not recommended for new deployments. Even as a mechanism for quickly testing Loki with your existing Beats/Logstash infrastructure we highly discourage the use of this plugin.

Reviewer: ```suggestion
Grafana Labs does not recommend using the Logstash plugin for new deployments. Even as a mechanism for quickly testing Loki with your existing Beats/Logstash infrastructure we highly discourage the use of this plugin.
```

Our experience over the years has found numerous significant challenges using Logstash and this plugin:

>   * It's very difficult to configure and get labels correct, conceptually Elasticsearch is a very different database from Loki and users almost always end up sending too many high cardinality labels to Loki which makes getting started with Loki unnecessarily complicated and confusing vs using other clients.

Reviewer: ```suggestion
  * It is very difficult to configure labels correctly.  Conceptually Elasticsearch is a very different database from Loki and users almost always end up sending too many high cardinality labels to Loki, which makes getting started with Loki unnecessarily complicated and confusing vs. using other clients.
```

>   * Logstash and the upstream Beats components implement backoff and flow control which we've found hard to observe leading to ingestion delays into Loki which are hard or impossible to address.

Reviewer: ```suggestion
  * Logstash and the upstream Beats components implement backoff and flow control which we've found hard to observe, leading to ingestion delays into Loki which are extremely difficult to address.
```

>   * We at Grafana Labs have no expertise at configuring Logstash or understanding its configuration language, so we can't help you with it.

Reviewer: ```suggestion
  * We at Grafana Labs have no expertise at configuring Logstash or understanding its configuration language, so we cannot provide support for it.
```

>   * It's very hard to troubleshoot and debug, our experience has shown that in nearly every case where it was assumed this would be the fast path to getting logs to Loki that was not the case and it ended up being a significant time sink.

Reviewer: ```suggestion
  * It's very hard to troubleshoot and debug. Our experience has shown that in nearly every case where it was assumed this would be the fast path to getting logs to Loki, that was not the case and it ended up taking far longer than anticipated.
```

  
Please... strongly consider using any alternative mechanism to sending logs to Loki.

We recommend using [Grafana Alloy](/docs/loki/latest/send-data/alloy/), this is the tool we build and the tool for which we can offer the best experience and most support.

> {{% /admonition %}}

Reviewer: ```suggestion
{{< /admonition >}}
```
